src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -22,75 +22,67 @@
 import net.sf.jabref.logic.importer.fileformat.MedlineImporter;
+import net.sf.jabref.logic.l10n.Localization;
 import net.sf.jabref.model.entry.BibEntry;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.apache.http.client.utils.URIBuilder;
 
 /**
- * Fetch or search from Pubmed http://www.ncbi.nlm.nih.gov/sites/entrez/
+ * Fetch or search from PubMed <a href=http://www.ncbi.nlm.nih.gov/sites/entrez/>www.ncbi.nlm.nih.gov</a>
+ * The MedlineFetcher fetches the entries from the PubMed database.
+ * See <a href=http://help.jabref.org/en/MedlineRIS>help.jabref.org</a> for a detailed documentation of the available fields.
  */
 public class MedlineFetcher implements IdBasedParserFetcher, SearchBasedFetcher {
 
     private static final Log LOGGER = LogFactory.getLog(MedlineFetcher.class);
 
     private static final Pattern ID_PATTERN = Pattern.compile("<Id>(\\d+)</Id>");
     private static final Pattern COUNT_PATTERN = Pattern.compile("<Count>(\\d+)<\\/Count>");
-    private static final Pattern RET_MAX_PATTERN = Pattern.compile("<RetMax>(\\d+)<\\/RetMax>");
-    private static final Pattern RET_START_PATTERN = Pattern.compile("<RetStart>(\\d+)<\\/RetStart>");
 
-    private int count;
-    private int retmax;
-    private int retstart;
+    private int numberOfResultsFound;
 
     /**
-     * Removes all comma's in a given String
+     * Replaces all commas in a given string with " AND "
      *
-     * @param query input to remove comma's
-     * @return input without comma's
+     * @param query input to remove commas
+     * @return input without commas
      */
     private static String replaceCommaWithAND(String query) {
         return query.replaceAll(", ", " AND ").replaceAll(",", " AND ");
     }
 
     /**
-     * When using 'esearch.fcgi?db=<database>&term=<query>' we will get a list of ID's matching the query.
+     * When using 'esearch.fcgi?db=<database>&term=<query>' we will get a list of IDs matching the query.
      * Input: Any text query (&term)
      * Output: List of UIDs matching the query
      *
      * @see <a href="https://www.ncbi.nlm.nih.gov/books/NBK25500/">www.ncbi.nlm.nih.gov/books/NBK25500/</a>
      */
-    private List<String> getPubMedIdsFromQuery(String query, int start, int pacing) throws FetcherException {
-        boolean doCount = true;
+    private List<String> getPubMedIdsFromQuery(String query) throws FetcherException {
         List<String> idList = new ArrayList<>();
         try {
-            URL ncbi = createSearchUrl(query, start, pacing);
+            URL ncbi = createSearchUrl(query);
             BufferedReader in = new BufferedReader(new InputStreamReader(ncbi.openStream()));
             String inLine;
+            //Everything relevant is listed before the IdList. So we break the loop right after the IdList tag closes.
             while ((inLine = in.readLine()) != null) {
-
+                if (inLine.contains("</IdList>")) {
```

Comment:
Uhoh, the service returns XML, but it is parsed using strings. @tschechlovdev Can you give us some indication about the effort for rewrite?